### PR TITLE
Mldb 1404 cls train dupe cols

### PR DESCRIPTION
**Pull request by mailletf**

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
@guyd please take a look at this in the context of our conversation earlier today :)

File: testing/MLDB-174-regression.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            "type": "classifier.train",
+            "params": {
+                "trainingData": { 
+                    "select": "{x} as features, y as label",
+                    "from": { "id": "test" }
+                },
+                "configuration": self.glz_conf,
+                "algorithm": "glz",
+                "modelFileUrl": modelFileUrl,
+                "equalizationFactor": 0.0,
```
Comment (guyd) [inline testing/MLDB-174-regression.py]: Is this still necessary and used for regressions?

File: testing/MLDB-174-regression.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+            "type": "classifier",
+            "params": {
+                "modelFileUrl": modelFileUrl
+            }
+        }
+
+        createFunctionOutput = mldb.put("/v1/functions/regressor", functionConfig)
+        mldb.log(createFunctionOutput.json())
+
+        selected = mldb.get("/v1/functions/regressor/application", input = { "features": { "x":10 }})
```
Comment (guyd) [inline testing/MLDB-174-regression.py]: Consider using a variable for the magic value here and below to make it clear that predicted y is supposed to be equal to x.

File: testing/MLDB-174-regression.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+                "modelFileUrlPattern": "file://tmp/MLDB-174-wine.cls",
+                "algorithm": "glz",
+                "configuration": self.glz_conf,
+                "mode": "regression",
+                "runOnCreation": True
+            }
+        }
+      
+        # start by training only with the pH and 'fixed acidity' features, specified
+        # excplicitely
```
Comment (guyd) [inline testing/MLDB-174-regression.py]: excplicitely -> explicitly

File: testing/MLDB-174-regression.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+            "params": {
+                "trainingData": """
+                    select
+                    {pH, "fixed acidity"} as features,
+                    quality as label
+                    from wine_red
+                """,
+                "datasetFolds": [
+                        {
+                            "training_where": "rowHash() % 2 = 0",
```
Comment (guyd) [inline testing/MLDB-174-regression.py]: It seems that there are still a couple of places where a configuration do not have the camel-case naming for fields.

File: testing/MLDB-174-regression.py [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+        selected = mldb.get("/v1/functions/regressor/application", input = { "features": { "x":10 }})
+
+        jsSelected = selected.json()
+        mldb.log(jsSelected)
+        result = jsSelected["output"]["score"]
+
+        self.assertAlmostEqual(result, 10)
+
+    def test_wine_quality_regression(self):
```
Comment (guyd) [inline testing/MLDB-174-regression.py]: This function does not test much of the regression code.  Consider using the scorer to test that wine quality in an expected range.

File: testing/MLDB-174-regression.py [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+                "modelFileUrlPattern": "file://tmp/MLDB-174-wine.cls",
+                "algorithm": "glz",
+                "configuration": self.glz_conf,
+                "mode": "regression",
+                "runOnCreation": True
+            }
+        }
+
+        # this should fail because we check for dupes
+        with self.assertRaises(mldb_wrapper.ResponseException) as re:
```
Comment (guyd) [inline testing/MLDB-174-regression.py]: Consider using the regex assert to check that your new message has the expected format and content.

File: testing/MLDB-174-regression.py [hunk 7]
```diff
@@ -0,0 +1,224 @@
+#
+# MLDB-184-regression.py
```
Comment (guyd) [inline testing/MLDB-174-regression.py]: 184 -> 174

File: testing/MLDB-174-regression.py [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+                "runOnCreation": True
+            }
+        }
+
+        # this should fail because we check for dupes
+        with self.assertRaises(mldb_wrapper.ResponseException) as re:
+            rez = mldb.put('/v1/procedures/wine_trainer', config)
+
+        
+        # let's recreate by avoiding collissions in rowNames
```
Comment (guyd) [inline testing/MLDB-174-regression.py]: ... and therefore duplicated columns.

**Comment by guyd** [comment]
+1 as long as regression test coverage is sufficient.

File: testing/MLDB-174-regression.py (hunk 1 above)
Comment (mailletf) [inline testing/MLDB-174-regression.py]: it is not used

**Pull request merged**